### PR TITLE
Update models.py

### PR DESCRIPTION
**Pull request by weissi1994**
Now showing also catchable_pokemons from API call result

## Motivation and Context
There were not enough Pokemons on my map 

## How Has This Been Tested?
- applied changes
- restarted workers
- lots of new pokemons on map

## Screenshots (if appropriate):
http://imgur.com/a/BlWkX

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

Now showing catchable_pokemons from api call result too

### CODE REVIEW

File: pogom/models.py [hunk 1]
```diff
@@ -324,6 +324,32 @@ def parse_map(map_dict, step_location):
...
+                }
+
+                webhook_data = {
+                    'encounter_id': b64encode(str(p['encounter_id'])),
+                    'spawnpoint_id': p['spawn_point_id'],
+                    'pokemon_id': p['pokemon_id'],
+                    'latitude': p['latitude'],
+                    'longitude': p['longitude'],
+                    'disappear_time': calendar.timegm(d_t.timetuple()),
+                    'last_modified_time': p['expiration_timestamp_ms'],
```
Comment (invisiblek) [inline pogom/models.py]: add
`'time_until_hidden_ms': p['time_till_hidden_ms'],`
like on 322?

**Comment by chuyskywalker** [comment]
I don't think this is doing what you think it's doing, or my area is too sparse for this to be effective. When I run this extra bit of code and output the `cell` data, all I'm seeing is that the `wild_pokemons` and `catchable_pokemons` are identical:

```
{
    'wild_pokemons': [{
        'last_modified_timestamp_ms': 1470669876934L,
        'longitude': -122.28404125707144,
        'pokemon_data': {'pokemon_id': 16},
        'latitude': 37.53225915641531,
        'spawn_point_id': u'808f9f3c23b',
        'encounter_id': 16663262614139451661L,
        'time_till_hidden_ms': 468038,
        }, {
        'last_modified_timestamp_ms': 1470669876934L,
        'longitude': -122.28430706520258,
        'pokemon_data': {'pokemon_id': 41},
        'latitude': 37.53202849072005,
        'spawn_point_id': u'808f9f3c269',
        'encounter_id': 7463643189922529453L,
        'time_till_hidden_ms': 794718,
        }],
    'catchable_pokemons': [{
        'pokemon_id': 16,
        'longitude': -122.28404125707144,
        'expiration_timestamp_ms': 1470670344972L,
        'latitude': 37.53225915641531,
        'spawn_point_id': u'808f9f3c23b',
        'encounter_id': 16663262614139451661L,
        }, {
        'pokemon_id': 41,
        'longitude': -122.28430706520258,
        'expiration_timestamp_ms': 1470670671652L,
        'latitude': 37.53202849072005,
        'spawn_point_id': u'808f9f3c269',
        'encounter_id': 7463643189922529453L,
        }],
    }
```
another scan hit
```			
{
    'wild_pokemons': [{
        'last_modified_timestamp_ms': 1470669881171L,
        'longitude': -122.28669930848693,
        'pokemon_data': {'pokemon_id': 16},
        'latitude': 37.533541275446495,
        'spawn_point_id': u'808f9f3d753',
        'encounter_id': 863368690703623613L,
        'time_till_hidden_ms': 705849,
        }, {
        'last_modified_timestamp_ms': 1470669881171L,
        'longitude': -122.28669930848693,
        'pokemon_data': {'pokemon_id': 16},
        'latitude': 37.533690805192116,
        'spawn_point_id': u'808f9f3d74b',
        'encounter_id': 10545559303519819085L,
        'time_till_hidden_ms': 606841,
        }],
    'catchable_pokemons': [{
        'pokemon_id': 16,
        'longitude': -122.28669930848693,
        'expiration_timestamp_ms': 1470670587020L,
        'latitude': 37.533541275446495,
        'spawn_point_id': u'808f9f3d753',
        'encounter_id': 863368690703623613L,
        }, {
        'pokemon_id': 16,
        'longitude': -122.28669930848693,
        'expiration_timestamp_ms': 1470670488012L,
        'latitude': 37.533690805192116,
        'spawn_point_id': u'808f9f3d74b',
        'encounter_id': 10545559303519819085L,
        }],
    }
```

**Comment by chuyskywalker** [comment]
Well, not "identical" but it's showing the same encounters.

**Comment by weissi1994** [comment]
ok, on my side the wild_pokemon array is always empty :O

**Comment by chuyskywalker** [comment]
Interesting; I suppose you could run through both dicts and put them all into the `pokemons` array that way we're not double webhook'ing. And any duplicates from the two lists would simply overwrite themselves.

**Comment by weissi1994** [comment]
maybe its because of my generated accounts?
The tos script from pgoapi/scripts has run over all accounts and they are all activated.

**Comment by chuyskywalker** [comment]
Maybe? Dunno really.

**Comment by tkSimon** [comment]
The long read: https://m.reddit.com/r/pokemongodev/comments/4v8rw4/whats_the_difference_between_catchable_and_wild/

**Comment by chuyskywalker** [comment]
Sounds like it is wide to merge them then

**Comment by chuyskywalker** [comment]
I think the code is going to need to look something more like:

```

    for cell in cells:
        if config['parse_pokemon']:

            for p in cell.get('catchable_pokemons', []):
                d_t = datetime.utcfromtimestamp(p['expiration_timestamp_ms'] / 1000.0)
                pokemons[p['encounter_id']] = {
                    'encounter_id': b64encode(str(p['encounter_id'])),
                    'spawnpoint_id': p['spawn_point_id'],
                    'pokemon_id': p['pokemon_id'],
                    'latitude': p['latitude'],
                    'longitude': p['longitude'],
                    'disappear_time': d_t
                }
            
            for p in cell.get('wild_pokemons', []):
                d_t = datetime.utcfromtimestamp(
                    (p['last_modified_timestamp_ms'] +
                     p['time_till_hidden_ms']) / 1000.0)
                pokemons[p['encounter_id']] = {
                    'encounter_id': b64encode(str(p['encounter_id'])),
                    'spawnpoint_id': p['spawn_point_id'],
                    'pokemon_id': p['pokemon_data']['pokemon_id'],
                    'latitude': p['latitude'],
                    'longitude': p['longitude'],
                    'disappear_time': d_t
                }
                
            # print and hook data
            for p in pokemons:
                printPokemon(p['pokemon_id'], p['latitude'], 
                             p['longitude'], p['disappear_time'])

                webhook_data = p
                webhook_data['is_lured'] = False
                send_to_webhook('pokemon', webhook_data)

        for f in cell.get('forts', []):
```

For a few reasons -- this way we don't double output the console print lines. Second, as of the last commit, this would have only been webhooking one set of the found pokemons.

The downside to what I wrote above is that the webhook is getting slightly less data -- mostly because `catchable` and `wild` have different data, so you gotta go with the least common denominator.

And while, not the fault of this pr, a webhook call PER POKEMON is bonkers; we should have a singular call with a list. I'll open a new bug about that sillyness.

**Comment by chuyskywalker** [comment]
I see you made it a list of pokemon to the webhook, we kinda need to not do that in this PR. Doing so would be a backwards incompatible change (a bit too much of one) and people have integrated against the webhooks as they are right now. Let's not break that.

**Comment by GamingMaster2000** [comment]
I get the following error when running the map with this code:

`2016-08-09 16:08:33,805 [ search_worker_4][        search][   ERROR] Exception in search_worker: 'long' object has no attribute '__getitem__'
Traceback (most recent call last):
  File "G:\GitHub\PokemonGo-Map\pogom\search.py", line 242, in search_worker_thread
    parsed = parse_map(response_dict, step_location)
  File "G:\GitHub\PokemonGo-Map\pogom\models.py", line 323, in parse_map
    printPokemon(p['pokemon_id'], p['latitude'],
TypeError: 'long' object has no attribute '__getitem__'`

File: pogom/models.py [hunk 2]
```diff
@@ -368,11 +368,11 @@ def parse_map(map_dict, step_location):
                     'last_modified': datetime.utcfromtimestamp(
                         f['last_modified_timestamp_ms'] / 1000.0),
                 }
-
+
```
Comment (invisiblek) [inline pogom/models.py]: whitespace

File: pogom/models.py [hunk 3]
```diff
@@ -368,11 +368,11 @@ def parse_map(map_dict, step_location):
                     'last_modified': datetime.utcfromtimestamp(
                         f['last_modified_timestamp_ms'] / 1000.0),
                 }
-
+                
     pokemons_upserted = 0
     pokestops_upserted = 0
     gyms_upserted = 0
-
+
```
Comment (invisiblek) [inline pogom/models.py]: whitespace

Comment (weissi1994) [inline pogom/models.py]: Landscape didnt nag about my whitespaces :/

**Comment by valmedia** [comment]
I got the following when testing this code:
2016-08-09 12:52:14,144 [ search_worker_1][        models][ WARNING] 'str' object has no attribute '_sort_key'... Retrying

**Comment by BaIthamel** [comment]
For feature requests please migrate this to the new feature page
http://feathub.com/PokemonGoMap/PokemonGo-Map/
Thankyou!

**Comment by CoolCanuck** [comment]
@BaIthamel This request needs to be opened again. FeatHub is no longer used. Good riddance.

**Pull request closed without merging**